tyro/_cli.py

```diff
@@ -345,4 +410,15 @@ def fix_arg(arg: str) -> str:
 
     if dummy_wrapped:
         out = getattr(out, _strings.dummy_field_name)
-    return out
+
+    # If we're parsed unknown args, we should return the original args, not
+    # the fixed ones.
+    if unknown_args is not None:
```

Comment:
There's something unsatisfying about how I set `unknown_args = None` in the case where `return_unknown_args` is set. Should probably be a better way to do this but as said before we need to know whether or not to `parse_args` or `parse_known_args`.